sdk/src/signer/keypair.rs

```diff
@@ -22,15 +22,15 @@ use {
 /// A vanilla Ed25519 key pair
 #[wasm_bindgen]
 #[derive(Debug)]
-pub struct Keypair(ed25519_dalek::Keypair);
+pub struct Keypair(ed25519_dalek::SigningKey);
```

Comment:
these aren't the same size.  seems like as good a time as any to abandon `Keypair`'s crappy interface

Reply:
fyi, i'm working in an alternative direction. sneak peek
```diff
diff --git a/sdk/src/signer/keypair.rs b/sdk/src/signer/keypair.rs
index c1e0803a6b..f247df3329 100644
--- a/sdk/src/signer/keypair.rs
+++ b/sdk/src/signer/keypair.rs
@@ -1,4 +1,5 @@
 #![cfg(feature = "full")]
+#![deprecated(since = "1.17.0", note = "it's time")]

 use {
     crate::{
```

Reply:
i have a side pr that restricts the `Keypair` APIs similarly to dalek 2.0.  i think these will be sufficient for 1.16.

i'm planning to totally deprecate the `Keypair` type from 1.17 forward and shift all interfaces to generics on our `Signer` trait.  there are a few potential complications here due to existing overly constrained APIs, but i don't believe they're widely used